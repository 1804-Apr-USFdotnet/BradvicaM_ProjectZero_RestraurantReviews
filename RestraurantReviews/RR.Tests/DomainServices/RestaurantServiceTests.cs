using ApprovalTests;
using ApprovalTests.Reporters;
using Autofac;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using RR.Console;
using RR.DomainContracts;
using RR.DomainServices;
using RR.Models;
using RR.RepositoryContracts;

namespace RR.Tests.DomainServices
{
    [TestClass]
    public class RestaurantServiceTests
    {
        private readonly Mock<IRestaurantRepository> _mock;

        public RestaurantServiceTests()
        {
            _mock = new Mock<IRestaurantRepository>();
            _mock.Setup(x => x.Add(It.IsAny<Restaurant>()));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TopThreeRestaurantByAverageReview_Returns_CorrectRestaurants()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.TopThreeRestaurantByAverageReview();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void AllRestaurants_Returns_AllRestaurantsFromDatabase()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.AllRestaurants();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void SearchByString_Returns_AllRestaurantsThatContainStringValue()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.SearchAll("Jordan");

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        public void AddRestaurant_Correctly_AddsRestuarantToDatabase()
        {
            var service = new RestaurantService(_mock.Object);

            var resturant = new Restaurant();

            service.AddRestaurant(resturant);

            _mock.Verify(x => x.Add(It.IsAny<Restaurant>()), Times.AtLeastOnce);
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void SearchByName_GivenString_ReturnsCorrectRestaurant()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var result = service.SearchByName("Billy Bobs Texas BBQ");

                Approvals.Verify(result);
            }
        }
    }
}
