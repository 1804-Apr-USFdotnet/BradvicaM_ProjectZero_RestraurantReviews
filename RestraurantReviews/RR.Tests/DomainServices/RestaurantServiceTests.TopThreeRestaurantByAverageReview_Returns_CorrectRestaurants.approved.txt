Restaurants[0] = 
Id: 7e9f9924-156e-4b9f-8a41-0f21137e5df4
Name: Yodel 
Street: 95612 Mayer Park
City: Salt Lake City
State: UT
ZipCode: 84170
PhoneNumber: 801 508 2699
Website: eventbrite.com
Rating: 10

Restaurants[1] = 
Id: c325cd17-4619-48bd-97c9-7b2e29db6de9
Name: Topicblab 
Street: 9407 Dorton Terrace
City: Ogden
State: UT
ZipCode: 84403
PhoneNumber: 801 302 4842
Website: auda.org.au
Rating: 10

Restaurants[2] = 
Id: cb76ceb5-3275-4e85-bf54-01059433d564
Name: Dabtype 
Street: 54894 Fisk Avenue
City: Cleveland
State: OH
ZipCode: 44191
PhoneNumber: 216 860 1827
Website: t.co
Rating: 8

